fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -4096,9 +4100,13 @@ struct StartFullRestoreTaskFunc : RestoreTaskFuncBase {
 		// Order does not matter, they will be put in order when written to the restoreFileMap below.
 		state std::vector<RestoreConfig::RestoreFile> files;
 
+		Version firstConsistentVersion = std::numeric_limits<int64_t>::min();
 		for (const RangeFile& f : restorable.get().ranges) {
 			files.push_back({ f.version, f.fileName, true, f.blockSize, f.fileSize });
+			firstConsistentVersion = std::max(firstConsistentVersion, f.version);
 		}
+		restore.firstConsistentVersion().set(tr, firstConsistentVersion);
+		wait(tr->commit());
```

Comment:
This is problematic when the `commit` fails, then the `firstConsistentVersion` is not set. In addition, `tr` is not reset, so I believe previous transaction's mutations are also sent to the database.

The easy fix is to move the first line into the next transaction, or have a transaction loop here, i.e.,
```
tr->reset()
loop { 
    try {
   ...
    } catch (Error& e) {
   ...
    }
}
```.

Reply:
Thanks. It doesn't make much sense to commit `firstConsistentVersion` with the next transaction as the next transaction commit is in a while loop. So I choose the second way.